scripts/linters/pre_commit_linter.py

```diff
@@ -0,0 +1,517 @@
+
+        Returns:
+            str. The data read from the file.
+        """
+        return self._get_data(filepath, mode)[0]
+
+    def readlines(self, filepath, mode='r'):
+        """Returns the tuple containing data line by line as read from the
+        file in unicode form.
+
+        Args:
+            filepath: str. The file path from which data is to be read.
+            mode: str. The mode in which the file is to be opened.
+
+        Returns:
+            tuple(str). The tuple containing data line by line as read from the
+                file.
+        """
+        return self._get_data(filepath, mode)[1]
+
+    def _get_data(self, filepath, mode):
+        """Returns the collected data from the file corresponding to the given
+        filepath.
+
+        Args:
+            filepath: str. The file path from which data is to be read.
+            mode: str. The mode in which the file is to be opened.
+
+        Returns:
+            tuple(str, tuple(str)). The tuple containing data read from the file
+                as first element and tuple containing the text line by line as
+                second element.
+        """
+        key = (filepath, mode)
+        if key not in self._CACHE_DATA_DICT:
+            with python_utils.open_file(filepath, mode) as f:
+                lines = f.readlines()
+                self._CACHE_DATA_DICT[key] = (''.join(lines), tuple(lines))
+        return self._CACHE_DATA_DICT[key]
+
+
+def _lint_all_files(
+        js_filepaths, ts_filepaths, py_filepaths, html_filepaths,
+        css_filepaths, file_extension_type, verbose_mode_enabled=False):
+    """Run all lint checks.
+
+    Args:
+        js_filepaths: list(str). The list of js filepaths to be linted.
+        ts_filepaths: list(str). The list of ts filepaths to be linted.
+        py_filepaths: list(str). The list of python filepaths to be linted.
+        html_filepaths: list(str). The list of HTML filepaths to be linted.
+        css_filepaths: list(str). The list of CSS filepaths to be linted.
+        verbose_mode_enabled: bool. True if verbose mode is enabled.
+        file_extension_type: list(str). The list of file extensions to be
+            linted.
+
+    Returns:
+        custom_linter: list. Custom lint checks.
+        third_party_linter: list. Third party lint checks.
+    """
+    parent_dir = os.path.abspath(os.path.join(os.getcwd(), os.pardir))
+    config_path_for_css_in_html = os.path.join(
+        parent_dir, 'oppia', '.stylelintrc')
+    config_path_for_oppia_css = os.path.join(
+        parent_dir, 'oppia', 'core', 'templates', 'css', '.stylelintrc')
+    custom_linters = []
+    third_party_linters = []
+
+    js_ts_file_extension = '.js' in file_extension_type or (
+        '.ts' in file_extension_type)
+    py_file_extension = '.py' in file_extension_type
+    html_file_extension = '.html' in file_extension_type
+    css_file_extension = '.css' in file_extension_type
```

Comment:
Do more  sanity-checking on inputs. If file_extension_type contains duplicates or bad inputs, call that out, explain what needs to be provided instead, and end early.

Better still -- if there is an option in argparse to allow only a limited subset of choices, use that to validate the input args.

Reply:
Changed to allow only a limited subset of choices and ignore the duplicate entry.